ec/resources/dnsroute/zone.go

```diff
@@ -10,1762 +10,574 @@ import (
-													Required: true,
-												},
-												"ip_version": {
-													Type:     schema.TypeInt,
-													Required: true,
-												},
-												"port_number": {
-													Type:     schema.TypeString,
-													Optional: true,
-												},
-												"reintegration_method_id": {
-													Type:     schema.TypeInt,
-													Required: true,
-												},
-												"status": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"status_name": {
-													Type:     schema.TypeString,
-													Optional: true,
-												},
-												"timeout": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"uri": {
-													Type:     schema.TypeString,
-													Required: true,
-												},
-											},
-										},
-									},
-									"weight": {
-										Type:     schema.TypeInt,
-										Required: true,
-									},
-									"record": {
-										Type: schema.TypeList,
-										Elem: &schema.Resource{
-											Schema: map[string]*schema.Schema{
-												"record_id": {
-													Type:     schema.TypeInt,
-													Computed: true,
-												},
-												"name": {
-													Type:     schema.TypeString,
-													Required: true,
-												},
-												"ttl": {
-													Type:     schema.TypeInt,
-													Required: true,
-												},
-												"rdata": {
-													Type:     schema.TypeString,
-													Required: true,
-												},
-												"verify_id": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"fixed_group_id": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"group_id": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"fixed_record_id": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"zone_id": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"fixed_zone_id": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"weight": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"record_type_id": {
-													Type:     schema.TypeInt,
-													Optional: true,
-												},
-												"record_type_name": {
-													Type:     schema.TypeString,
-													Optional: true,
-												},
-												"is_delete": {
-													Type:     schema.TypeBool,
-													Optional: true,
-												},
-											},
-										},
-										Required: true,
-									},
-								},
-							},
+							Elem:     groupRecordSchema,
+							Description: `Defines a set of CNAME records 
+							associated with this group.`,
```

Comment:
Great work going through this huge schema... it's so long I wonder if it might be worth moving it into a `zone_schema.go` file to make it easier to work with the CRUD methods

Reply:
@stevenpaz added a backlog item. I'd like to do this and add some reuse with group.go that also has a similar schema definition. First attempt caused multiple issues with Terraform diffs so I'd like to tackle this separately.